File: singlequiz/QuestionViewController.m
//
//  QuestionViewController.m
//  singlequiz
//
//  Created by KHUN NINE on 8/21/15.
//  Copyright (c) 2015 KHUN NINE. All rights reserved.
//

#import "QuestionViewController.h"
#import "ResultViewController.h"
#import "DataController.h"

#import "Extension.h"
#import "singlequiz-Swift.h"

@interface QuestionViewController ()
@end

@implementation QuestionViewController

int questionNo = 0;

UIImageView *resultImageView;

UILabel *question;
NSMutableArray *choicesList;

//    // MARK: - View life cycle
- (void)viewDidLoad {
    self.view.backgroundColor = [UIColor clearColor];
}

- (void)viewWillAppear:(BOOL)animated {
    NSString *screenName = [NSString stringWithFormat: @"iOS Question view no. %d", questionNo];
    [UserLogged trackScreen: screenName];
}

// MARK: - Set and generate choices
- (void) generateChoices: (CGFloat) position {

    const CGFloat buttonWidth = CGRectGetMaxX( self.view.frame ) * 0.75;
    const CGFloat buttonHeight = ([[self iPhoneScreenSize] isEqual: @"3.5"]) ? 60 : 66;

    const CGFloat margin = 8;
    const CGFloat buttonMargin = buttonHeight + margin/3;
//
    for (int i=0 ; i<choicesList.count ; i++) {
        CGFloat yPosition = position + (buttonMargin * i);
        CGRect  frame = CGRectMake(0, yPosition, buttonWidth, buttonHeight);
        
        UIButton *button = [self choiceButtonMake:i withFrame:frame];
        
        [self.view addSubview:button];
    }
}

- (UIButton *) choiceButtonMake:(int) index withFrame:(CGRect) frame {
    
    
    UIButton *button = [[UIButton alloc]initWithFrame:frame];
    button.titleLabel.font = [UIFont fontWithName: @"SukhumvitSet-Medium" size:13.0];
    button.titleLabel.lineBreakMode = NSLineBreakByWordWrapping;
    button.titleLabel.textAlignment = NSTextAlignmentCenter;
    button.center = CGPointMake(self.view.center.x, button.center.y);
    button.layer.cornerRadius = 8;
    button.layer.borderColor = [UIColor lightGrayColor].CGColor;
    button.layer.borderWidth = 1;
    button.tag = index;
    [button setTitle:choicesList[index] forState:UIControlStateNormal];
    [button setTitleColor:[UIColor colorWithWhite:0.2 alpha:1] forState: UIControlStateNormal];
    [button setTitleColor:[UIColor colorWithWhite:0.5 alpha:1] forState:UIControlStateHighlighted];
    [button addTarget:self action:@selector(choiceButtonClick:) forControlEvents:UIControlEventTouchUpInside];
    
    return button;
}

- (void) choiceButtonClick:(UIButton *) sender {

    [DataController sharedInstance].summation = [DataController sharedInstance].summation + (sender.tag + 1);

        if (questionNo == [DataController sharedInstance].questions.count - 1) {
            
            [SwiftSpinner show:@"กำลังประมวลผล" animated:true];
            
            CGFloat delay = 3 * NSEC_PER_SEC;
            NSTimeInterval time = dispatch_time(DISPATCH_TIME_NOW, delay);
            
            dispatch_after(time, dispatch_get_main_queue(), ^{
                
                [SwiftSpinner hideWithCompletion: ^{
                    ResultViewController *viewController = [ResultViewController alloc];
                    viewController.modalTransitionStyle = UIModalTransitionStyleFlipHorizontal;
                    [self presentViewController:viewController animated:true completion:nil];
                }];
            });
        }
        else {
            
            QuestionViewController *viewController = [DataController sharedInstance].list_questionViewController[questionNo + 1];
            viewController.modalTransitionStyle = UIModalTransitionStyleFlipHorizontal;
            [self presentViewController:viewController animated:true completion:nil];
        }
}

//    
//    // MARK: - Status bar
- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (BOOL)prefersStatusBarHidden {
    return true;
}

//    // MARK: - Setter methods
- (void) setQuestionNumber: (int) number {
    questionNo = number;
}

- (void) setQuestionPhoto {
    NSString *imageString = [NSString stringWithFormat: @"q%d.jpg", questionNo+1];
    UIImage  *image       = [UIImage imageNamed:imageString];
    
    resultImageView = [[UIImageView alloc] initWithImage:image];
    
    if ([[self iPhoneScreenSize] isEqual: @"3.5"]) {
        resultImageView.frame = CGRectMake(0, 0, CGRectGetMidX(self.view.frame)-30, CGRectGetMidX(self.view.frame)-30);
        resultImageView.center = CGPointMake(CGRectGetMidX(self.view.frame), CGRectGetMaxY(self.view.frame) * 0.18);
    }
    else if ([[self iPhoneScreenSize] isEqual: @"4"]) {
        resultImageView.frame = CGRectMake(0, 0, CGRectGetMidX(self.view.frame)-10, CGRectGetMidX(self.view.frame)-10);
        resultImageView.center = CGPointMake(CGRectGetMidX(self.view.frame), CGRectGetMaxY(self.view.frame) * 0.18);
    }
    else {
        resultImageView.frame = CGRectMake(0, 0, CGRectGetMidX(self.view.frame)-10, CGRectGetMidX(self.view.frame)-10);
        resultImageView.center = CGPointMake(CGRectGetMidX(self.view.frame), CGRectGetMaxY(self.view.frame) * 0.225);
    }
    
    resultImageView.layer.zPosition    = 3;
    resultImageView.layer.borderWidth  = 4;
    resultImageView.layer.borderColor  = [UIColor whiteColor].CGColor;
    resultImageView.layer.cornerRadius = resultImageView.frame.size.width/2;
    resultImageView.contentMode = UIViewContentModeScaleAspectFill;
    resultImageView.clipsToBounds = true;
    
    [self.view addSubview:resultImageView];
}

- (void) setQuestionLabel: (NSString *) questionText {
    
    UILabel *question = [[UILabel alloc]initWithFrame:CGRectMake(0, 0, CGRectGetMaxX( self.view.frame ) * 0.8, 100)];
        question.font = [UIFont fontWithName: @"SukhumvitSet-Medium" size:18.0];
    question.center = CGPointMake(self.view.center.x, self.view.frame.size.height * 0.435);
    question.numberOfLines = 0;
    question.textAlignment = NSTextAlignmentCenter;
    question.text = questionText;
    
    if ([[self iPhoneScreenSize] isEqual: @"3.5"]) {
        question.center = CGPointMake(self.view.center.x, self.view.frame.size.height * 0.39);
    }
    else if ([[self iPhoneScreenSize] isEqual: @"4"]) {
        question.center = CGPointMake(self.view.center.x, self.view.frame.size.height * 0.40);
    }
    
    [self.view addSubview:question];
}

- (void) addChoice: (NSString *) newChoice {
    
    [choicesList addObject:newChoice];
}


- (void) setChoicesLabels {

        
    CGFloat yPositionFirstChoice;
        
    if ([[self iPhoneScreenSize] isEqualToString:@"3.5"]) {
        yPositionFirstChoice = self.view.frame.size.height * 0.45;
    }
    else if ([[self iPhoneScreenSize] isEqualToString:@"3.5"]) {
        yPositionFirstChoice = self.view.frame.size.height * 0.47;
    }
    
    [self generateChoices:yPositionFirstChoice];
}

- (void) setContentBackgroundImageView {
    
    NSString *imageString = [NSString stringWithFormat: @"q%d.jpg", questionNo+1];
    UIImage  *image       = [UIImage imageNamed:imageString];
    
    UIImageView *imageView = [[UIImageView alloc] initWithImage:image];
    imageView.frame = CGRectMake(0, 0, self.view.frame.size.width - 16, CGRectGetMaxY(self.view.frame) - 16);
    imageView.layer.cornerRadius = 0;
    imageView.contentMode = UIViewContentModeScaleAspectFill;
    imageView.clipsToBounds = true;
    
    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
    UIVisualEffectView *blurView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
    blurView.frame = imageView.frame;
    [blurView setTranslatesAutoresizingMaskIntoConstraints:false];
        
        // Set image over layer
    CAGradientLayer *gradient = [[CAGradientLayer alloc] initWithLayer:imageView.layer];
    gradient.frame = imageView.frame;
        
        // Add colors to layer
    UIColor *startColor = [UIColor colorWithRed:1 green:1 blue:1 alpha:0.4];
    UIColor *endColor   = [UIColor colorWithRed:1 green:1 blue:1 alpha:0.2];
    gradient.colors = [NSArray arrayWithObjects:(id)[startColor CGColor], (id)[endColor CGColor], nil];
        
        UIImageView *clipper = [[UIImageView alloc]initWithFrame: CGRectMake(0,
                                                                           0,
                                                                           self.view.frame.size.width  - 16,
                                                                           self.view.frame.size.height - 16)];
    [clipper addSubview:imageView];
    [clipper addSubview:blurView];
    [clipper.layer addSublayer:gradient];
    clipper.clipsToBounds = true;
    clipper.layer.cornerRadius = 8;
        
    [self.view addSubview:clipper];
}


+ (void) setQuestionViewControllers {
    
    NSArray *questions   = [DataController sharedInstance].questions;
    NSArray *choicesList = [DataController sharedInstance].choicesList;
    
    NSLog(@"%d", [DataController sharedInstance].choicesList.count);
    
    CGFloat numberOfChoicePerQuestion = [DataController sharedInstance].choicesList.count / [DataController sharedInstance].questions.count;
    CGFloat numberOfQuestion = [DataController sharedInstance].questions.count;
    
    for (int i=0 ; i<numberOfQuestion ; i++) {
        
        QuestionViewController *questionViewController = [QuestionViewController alloc];
        
        for (int j=0 ; j<numberOfChoicePerQuestion ; j++) {
            
            NSString *choice = [choicesList objectAtIndex: (i * numberOfChoicePerQuestion + j)];
            [questionViewController addChoice:choice];
        }
        
        [questionViewController setQuestionNumber:i];
        [questionViewController setContentBackgroundImageView];
        [questionViewController setQuestionLabel:questions[i]];
        [questionViewController setChoicesLabels];
        [questionViewController setQuestionPhoto];
        
        // NEED TO BE FIXED
        // DataController.list_questionViewController. append( questionViewController )
    }

}

// MARK: - Setter methods and controllers
+ (void) getStartedQuestion:(UIViewController *) rootView {

    [DataController sharedInstance].summation = [DataController sharedInstance].result;

    CGFloat delay = 3 * NSEC_PER_SEC;
    NSTimeInterval time = dispatch_time(DISPATCH_TIME_NOW, delay);
    
    dispatch_after(time, dispatch_get_main_queue(), ^{
        
        [self presentFirstQuestion:rootView];
    });
}

+ (void) presentFirstQuestion: (UIViewController*) rootView {

    NSLog(@"%lu",(unsigned long)[DataController sharedInstance].list_questionViewController.count);
    QuestionViewController *firstQuestionView  = [DataController sharedInstance].list_questionViewController[0];
    firstQuestionView.modalTransitionStyle = UIModalTransitionStyleCoverVertical;
    [rootView presentViewController:firstQuestionView animated:true completion:nil];
}

// MARK: - Controller methods
- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {

    UITouch *touch = [[event touchesForView:self.view] anyObject];
    CGPoint point  = [touch locationInView:touch.view];
}


@end

File: singlequiz/QuestionViewController.swift
//
//  QuestionViewController.swift
//  WhyDoPeopleLikeYou
//
//  Created by KHUN NINE on 6/30/15.
//  Copyright (c) 2015 KHUN NINE. All rights reserved.
//

import Foundation
import UIKit

@objc class QuestionViewController: UIViewController {
    
    // MARK: - Variables
    let margin: CGFloat = 8
    let elementHeight: CGFloat = 44
    
    var questionNo: Int = 0
    
    var resultImageView: UIImageView!
    
    var question: UILabel!
    var choices = [String]()
    var choiceView: UIView!

    // MARK: - View life cycle
    override func viewDidLoad() {
        view.backgroundColor = UIColor.clearColor()
    }
    
    override func viewWillAppear(animated: Bool) {
        UserLogged.trackScreen("Question view: \(questionNo)")
    }

    // MARK: - Set and generate choices
    func generateChoices (position: CGFloat) {
        
        let buttonWidth  : CGFloat = CGRectGetMaxX( self.view.frame ) * 0.75
        let buttonHeight : CGFloat = (UIViewController().iPhoneScreenSize() == "3.5") ? 60 : 66
        
        let margin: CGFloat = 8
        let buttonMargin: CGFloat = buttonHeight + margin/3
        
        for (var i=0 ; i<choices.count ; i++) {
            
            var yPosition = position + (buttonMargin * CGFloat(i))
            var frame = CGRectMake(0, yPosition, buttonWidth ,buttonHeight )
            
            var button = choiceButtonMake(i, frame: frame)
            
            self.view.addSubview(button)
            
        }
    }
    
    func choiceButtonMake (index: Int, frame: CGRect) -> UIButton {
        
        var button = UIButton(frame: frame)
        button.titleLabel?.font = UIFont(name: "SukhumvitSet-Medium", size: 15)
        button.titleLabel?.lineBreakMode = NSLineBreakMode.ByWordWrapping
        button.titleLabel?.textAlignment = NSTextAlignment.Center
        button.center.x = self.view.center.x
        button.setTitleColor(UIColor(white: 0.2, alpha: 1), forState: .Normal)
        button.setTitleColor(UIColor(white: 0.5, alpha: 1), forState: .Highlighted)
        button.backgroundColor = UIColor(white: 1, alpha: 1)
        button.layer.cornerRadius = 8
        button.layer.borderColor = UIColor.lightGrayColor().CGColor
        button.layer.borderWidth = 1
        button.setTitle( choices[index], forState: .Normal)
        button.addTarget(self, action: "choiceButtonClick:", forControlEvents: UIControlEvents.TouchUpInside)
        button.tag = index

        return button
    }
    
    func choiceButtonClick (sender: UIButton!) {
        
        DataController.summation = DataController.summation + (sender.tag + 1)
        //println(DataController.summation)
        
        if (questionNo == DataController.questions.count - 1) {
            
            SwiftSpinner.show("กำลังประมวลผล", animated: true)
            
            let delay = 3 * Double(NSEC_PER_SEC)
            let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
            dispatch_after(time, dispatch_get_main_queue()) {
                
                SwiftSpinner.hide(completion: {
                    
                    var viewController = ResultViewControllers()
                    viewController.modalTransitionStyle = UIModalTransitionStyle.FlipHorizontal
                    self.presentViewController(viewController, animated: true, completion: nil)
                })
                
            }
        }
        else {
            
            var viewController = DataController.list_questionViewController[ questionNo + 1 ]
            viewController.modalTransitionStyle = UIModalTransitionStyle.FlipHorizontal
            self.presentViewController(viewController, animated: true, completion: nil)
        }
    }

    // MARK: - Status bar
    override func preferredStatusBarStyle() -> UIStatusBarStyle {
        return UIStatusBarStyle.LightContent
    }
    
    override func prefersStatusBarHidden() -> Bool {
        return true
    }

    // MARK: - Setter methods
    func setQuestionNumber (number: Int) {
        self.questionNo = number
    }
    
    func setQuestionPhoto () {
        
        var imageString = "q\(questionNo+1).jpg"
        var image = UIImage(named: imageString)
        
        resultImageView = UIImageView(image: image)
        
        if (UIViewController().iPhoneScreenSize() == "3.5") {
            resultImageView.frame = CGRect(x: 0, y: 0, width: CGRectGetMidX(self.view.frame)-30, height: CGRectGetMidX(self.view.frame)-30)
            resultImageView.center.y = CGRectGetMaxY(self.view.frame) * 0.18
        }
        else if (UIViewController().iPhoneScreenSize() == "4") {
            resultImageView.frame = CGRect(x: 0, y: 0, width: CGRectGetMidX(self.view.frame)-10, height: CGRectGetMidX(self.view.frame)-10)
            resultImageView.center.y = CGRectGetMaxY(self.view.frame) * 0.18
        }
        else {
            resultImageView.frame = CGRect(x: 0, y: 0, width: CGRectGetMidX(self.view.frame)-10, height: CGRectGetMidX(self.view.frame)-10)
            resultImageView.center.y = CGRectGetMaxY(self.view.frame) * 0.225
        }
        
        resultImageView.center.x = CGRectGetMidX(self.view.frame)
        resultImageView.layer.zPosition    = 3
        resultImageView.layer.borderWidth  = 4
        resultImageView.layer.borderColor  = UIColor.whiteColor().CGColor
        resultImageView.layer.cornerRadius = resultImageView.frame.width/2
        resultImageView.contentMode = UIViewContentMode.ScaleAspectFill
        resultImageView.clipsToBounds = true
        
        self.view.addSubview( resultImageView )
    }
    
    func setQuestionLabel (questionText: String) {
        
        question = UILabel(frame: CGRectMake(0, 0, CGRectGetMaxX( self.view.frame ) * 0.8, 100))
        question.font = UIFont(name: "SukhumvitSet-Medium", size: 18)
        question.center.x = self.view.center.x
        question.center.y = self.view.frame.height * 0.435
        question.numberOfLines = 0
        question.textAlignment = NSTextAlignment.Center
        question.text = questionText
        
        if (UIViewController().iPhoneScreenSize() == "3.5") {
            question.center.y = self.view.frame.height * 0.39
        }
        else if (UIViewController().iPhoneScreenSize() == "4") {
            question.center.y = self.view.frame.height * 0.40
        }
        
        self.view.addSubview(question)
    }
    
    
    func addChoices (newChoice: String) {
        
        choices.append( newChoice )
    }
    
    func setChoicesLabels () {
        
        var yPositionFirstChoice: CGFloat!
        
        switch UIViewController().iPhoneScreenSize() {
            
            case "3.5":
            yPositionFirstChoice = self.view.frame.height * 0.45
            
            case "4":
            yPositionFirstChoice = self.view.frame.height * 0.47
            
            default:
            yPositionFirstChoice = self.view.frame.height * 0.5
        }
        
        generateChoices( yPositionFirstChoice )
    }
    
    func setContentBackgroundImageView () {
        
        setContentBackgroundImage_Style1()
        // Uncomment the like below to change the content background image style
        // setContentBackgroundImage_Style2()
    }
    
    func setContentBackgroundImage_Style2 () {
        
         setContentBackgroundImage()
         setCoverBackgroundImage()
         setBlurBackgroundImage()
    }
    
    func setContentBackgroundImage_Style1 () {
        
        var imageView = UIImageView(image: UIImage(named: "q\(questionNo+1).jpg"))
        imageView.frame = CGRect(x: 0, y: 0, width: self.view.frame.width - margin * 2, height: CGRectGetMaxY(self.view.frame) - margin*2)
        imageView.layer.cornerRadius = 0
        imageView.contentMode = UIViewContentMode.ScaleAspectFill
        imageView.clipsToBounds = true
        
        let blurEffect: UIBlurEffect = UIBlurEffect(style: .Light)
        let blurView: UIVisualEffectView = UIVisualEffectView(effect: blurEffect)
        blurView.frame = imageView.frame
        blurView.setTranslatesAutoresizingMaskIntoConstraints(false)
        
        // Set image over layer
        var gradient: CAGradientLayer = CAGradientLayer(layer: imageView.layer)
        gradient.frame = imageView.frame;
        
        // Add colors to layer
        var startColor: UIColor   = UIColor(red: 1, green: 1, blue: 1, alpha: 0.4)
        var endColor: UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0.2)
        gradient.colors = NSArray(objects: startColor.CGColor, endColor.CGColor) as [AnyObject]
        
        var clipper = UIImageView(frame: CGRect(x: margin, y: margin, width: self.view.frame.width - margin * 2, height: self.view.frame.height - margin*2))
        clipper.layer.cornerRadius = 8
        clipper.addSubview(imageView)
        clipper.addSubview(blurView)
        clipper.layer.addSublayer(gradient)
        clipper.clipsToBounds = true
        
        self.view.addSubview( clipper )
    }
    
    func setContentBackgroundImage () {
        
        // defind the top margin
        let statusBarHeight: CGFloat = 20
        let topMargin: CGFloat = margin
        
        // Create a shape
        var contentBackgroundImageShape = CAShapeLayer()
        contentBackgroundImageShape.frame = CGRect(x: margin, y: margin, width: self.view.frame.width - margin * 2, height: self.view.frame.height - ( margin + topMargin))
        contentBackgroundImageShape.path = UIBezierPath(roundedRect: contentBackgroundImageShape.bounds, cornerRadius: 6).CGPath
        contentBackgroundImageShape.fillColor = UIColor(white: 1, alpha: 1).CGColor
        contentBackgroundImageShape.strokeColor = UIColor.grayColor().CGColor
        contentBackgroundImageShape.lineWidth = 0.3;
        
        self.view.layer.addSublayer( contentBackgroundImageShape )
        
    }
    
    func setCoverBackgroundImage () {
        
        // Set background image view for each question
        var imageView = UIImageView(image: UIImage(named: "q\(questionNo+1).jpg"))
        imageView.frame = CGRect(x: 0, y: 0, width: self.view.frame.width - margin * 2, height: CGRectGetMaxY(self.view.frame) * 0.25 - margin)
        imageView.layer.cornerRadius = 0
        imageView.contentMode = UIViewContentMode.ScaleAspectFill
        imageView.clipsToBounds = true
        
        // Set blur effect to image view
        let blurEffect: UIBlurEffect = UIBlurEffect(style: .Light)
        let blurView: UIVisualEffectView = UIVisualEffectView(effect: blurEffect)
        blurView.frame = imageView.frame
        blurView.setTranslatesAutoresizingMaskIntoConstraints(false)
        
        // Clips image view to bounds
        var clipper = UIImageView(frame: CGRect(x: margin, y: margin, width: self.view.frame.width - margin * 2, height: self.view.frame.height - margin))
        clipper.layer.cornerRadius = 8
        clipper.addSubview(imageView)
        clipper.addSubview(blurView)
        clipper.clipsToBounds = true
        
        self.view.addSubview( clipper )
    }
    
    func setBlurBackgroundImage () {
        
        // Set background image view for each question
        var imageView = UIImageView(image: UIImage(named: "q\(questionNo+1).jpg"))
        imageView.frame = CGRect(x: 0, y: CGRectGetMaxY(self.view.frame) * 0.25 - margin, width: self.view.frame.width - margin * 2, height: CGRectGetMaxY(self.view.frame) - CGRectGetMaxY(self.view.frame) * 0.25 - margin)
        imageView.layer.cornerRadius = 0
        imageView.contentMode = UIViewContentMode.ScaleAspectFill
        imageView.clipsToBounds = true
        
        // Set blur effect to image view
        let blurEffect: UIBlurEffect = UIBlurEffect(style: .ExtraLight)
        let blurView: UIVisualEffectView = UIVisualEffectView(effect: blurEffect)
        blurView.frame = imageView.frame
        blurView.setTranslatesAutoresizingMaskIntoConstraints(false)
        
        // Set image over layer
        var gradient: CAGradientLayer = CAGradientLayer(layer: imageView.layer)
        gradient.frame = imageView.frame;
        
        // Add colors to layer
        var startColor:   UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
        var fadeInColor:  UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 1)
        var centerColor:  UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0.5)
        var fadeOutColor: UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0.1)
        var endColor:     UIColor = UIColor(red: 1, green: 1, blue: 1, alpha: 0)
        gradient.colors = NSArray(objects: startColor.CGColor, fadeInColor.CGColor, centerColor.CGColor, fadeOutColor.CGColor, endColor.CGColor) as [AnyObject]
        
        var clipper = UIImageView(frame: CGRect(x: margin, y: margin, width: self.view.frame.width - margin * 2, height: self.view.frame.height - margin))
        clipper.layer.addSublayer( gradient )
        clipper.layer.cornerRadius = 8
        clipper.clipsToBounds = true
        clipper.addSubview(imageView)
        clipper.addSubview(blurView)
        
        self.view.addSubview( clipper )
    }
    
    class func setQuestionViewControllers () {
        
        let questions = DataController.questions
        let choices   = DataController.choices
        
        let numberOfChoicePerQuestion = DataController.choices.count/DataController.questions.count
        let numberOfQuestion = DataController.questions.count
        
        for (var i=0 ; i<numberOfQuestion ; i++) {
            
            var questionViewController = QuestionViewController()
            
            for (var j=0 ; j<numberOfChoicePerQuestion ; j++) {
                
                let choice = choices[i * numberOfChoicePerQuestion + j]
                questionViewController.addChoices(choice)
            }
            
            questionViewController.setQuestionNumber(i)
            questionViewController.setContentBackgroundImageView()
            questionViewController.setQuestionLabel(questions[i])
            questionViewController.setChoicesLabels()
            questionViewController.setQuestionPhoto()
            
            DataController.list_questionViewController.append( questionViewController )
        }
    }

    // MARK: - Setter methods and controllers
    class func getStartedQuestion (rootView: UIViewController) {
        
        DataController.summation = DataController.result
        
        let timeDelay: Double = 0
        let delay = timeDelay * Double(NSEC_PER_SEC)
        let time = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
        
        dispatch_after(time, dispatch_get_main_queue()) {
            self.presentFirstQuestion(rootView)
        }
    }
    
    class func presentFirstQuestion (rootView: UIViewController) {
        
        let questionView  = DataController.list_questionViewController
        var firstQuestion = questionView.first
        firstQuestion!.modalTransitionStyle = UIModalTransitionStyle.CoverVertical
        rootView.presentViewController( firstQuestion!, animated: true, completion: nil)
    }

    // MARK: - Controller methods
    override func touchesBegan(touches: Set<NSObject>, withEvent event: UIEvent) {
        var touch = touches.first as! UITouch
        var point = touch.locationInView(self.view)
    }
}